alt-fuel-finder/app/models/station.rb

```diff
@@ -1,6 +1,34 @@
 class Station
+  attr_reader :service
+  Station = Struct.new("Station",
+                      :id,
+                      :fuel_type_code,
+                      :station_phone,
+                      :city,
+                      :intersection_directions,
+                      :street_address,
+                      :fuel_type_code,
+                      :access_days_time,
+                      :distance
+                      )
 
-  def initialize(params)
+  def self.service
+    @service ||= NrelService.new
```

Comment:
@josh-works I like the attempt to use memoization here. One thing to note is this is an instance variable being used at the class level. This can in some situations lead to unexpected behavior. I tend to use memoization for things like this but at the instance method layer instead.